src/v2i-hub/ImmediateForwardPlugin/src/ImmediateForwardPlugin.cpp

```diff
@@ -270,172 +234,178 @@ bool ImmediateForwardPlugin::ParseJsonMessageConfig(const std::string& json, uin
 
 
-			// Send the message using the configured UDP client.
 
-			for (uint i = 0; i < _udpClientList[_messageConfigMap[configIndex].ClientIndex].size(); i++)
-			{
-				//cout << message << endl;
+				// Send the message using the configured UDP client.
 
-				if (_udpClientList[_messageConfigMap[configIndex].ClientIndex][i] != NULL)
+				for (uint i = 0; i < _udpClientList[_messageConfigMap[configIndex].ClientIndex].size(); i++)
 				{
-					PLOG(logDEBUG2) << _logPrefix << "Sending - TmxType: " << _messageConfigMap[configIndex].TmxType << ", SendType: " << _messageConfigMap[configIndex].SendType
-						<< ", PSID: " << _messageConfigMap[configIndex].Psid << ", Client: " << _messageConfigMap[configIndex].ClientIndex
-						<< ", Channel: " << (_messageConfigMap[configIndex].Channel.empty() ? ::to_string( msg->dsrcMetadata->channel) : _messageConfigMap[configIndex].Channel)
-						<< ", Port: " << _udpClientList[_messageConfigMap[configIndex].ClientIndex][i]->GetPort();
+					//cout << message << endl;
 
-					_udpClientList[_messageConfigMap[configIndex].ClientIndex][i]->Send(message);
-				}
-				else
-				{
-					SetStatus<uint>(Key_SkippedInvalidUdpClient, ++_skippedInvalidUdpClient);
-					PLOG(logWARNING) << "UDP Client Invalid. Cannot send message. TmxType: " << _messageConfigMap[configIndex].TmxType;
+					if (_udpClientList[_messageConfigMap[configIndex].ClientIndex][i] != NULL)
+					{
+						PLOG(logDEBUG2) << _logPrefix << "Sending - TmxType: " << _messageConfigMap[configIndex].TmxType << ", SendType: " << _messageConfigMap[configIndex].SendType
+							<< ", PSID: " << _messageConfigMap[configIndex].Psid << ", Client: " << _messageConfigMap[configIndex].ClientIndex
+							<< ", Channel: " << (_messageConfigMap[configIndex].Channel.empty() ? ::to_string( msg->dsrcMetadata->channel) : _messageConfigMap[configIndex].Channel)
+							<< ", Port: " << _udpClientList[_messageConfigMap[configIndex].ClientIndex][i]->GetPort();
+
+						_udpClientList[_messageConfigMap[configIndex].ClientIndex][i]->Send(message);
+					}
+					else
+					{
+						SetStatus<uint>(Key_SkippedInvalidUdpClient, ++_skippedInvalidUdpClient);
+						PLOG(logWARNING) << "UDP Client Invalid. Cannot send message. TmxType: " << _messageConfigMap[configIndex].TmxType;
+					}
 				}
 			}
 		}
-	}
-	if (!foundMessageType)
-	{
-		SetStatus<uint>(Key_SkippedNoMessageRoute, ++_skippedNoMessageRoute);
-		PLOG(logWARNING)<<" WARNING TMX Subtype not found in configuration. Message Ignored: " <<
-				"Type: " << msg->type << ", Subtype: " << msg->subtype;
-		return;
-	}
+		if (!foundMessageType)
+		{
+			SetStatus<uint>(Key_SkippedNoMessageRoute, ++_skippedNoMessageRoute);
+			PLOG(logWARNING)<<" WARNING TMX Subtype not found in configuration. Message Ignored: " <<
+					"Type: " << msg->type << ", Subtype: " << msg->subtype;
+			return;
+		}
 
 
-}
+	}
+
 
 
 } /* namespace ImmediateForward */
+// The main entry point for this application.
+int main(int argc, char *argv[])
```

Comment:
Not sure why we prefer to relocate this function.

Reply:
See comment above